### ISSUE TITLE
Export bug when time-dependent variable is all 0.

### ISSUE THREAD

**Issue by fneum** [PyPSA/PyPSA#722]
## Checklist

- [x] I am using the current [`master`](https://github.com/pypsa/pypsa/tree/master) branch.

## Describe the Bug

When a time-dependent variable, in this case n.links_t.p2 is all 0., the whole table is dropped upon export, here is the line of code where it happens:

https://github.com/PyPSA/PyPSA/blob/a0027f50a24744251e58ea6577f446bfdc90f1f6/pypsa/io.py#L504

I.e. if all columns equal the default, the whole table is dropped. Which leads to results disappearing.

## MWE

```py
!wget https://model.energy/networks/c6c1d08a805dbbeae283ce07a1195934.nc
import pypsa
n = pypsa.Network("c6c1d08a805dbbeae283ce07a1195934.nc")
(n.links_t.p2 != n.components["Link"]["attrs"].at["p2","default"]).any()
n.links_t.p2["hydrogen_electrolyser"] = 0.
n.export_to_netcdf("c6-test.nc")
n = pypsa.Network("c6-test.nc")
n.links_t.p2
n.links_t.p2["hydrogen_compressor"] = 0.
n.export_to_netcdf("c6c-test.nc")
n = pypsa.Network("c6c-test.nc")
n.links_t.p2
```

## Solution

Should not actually drop anything, i.e. save the file exactly as the user expects, then let the compression algorithm worry about saving disk space.

**Comment by coroa** [comment]
I agree this makes sense for the netcdf representation, where there is compression (which is now luckily includede), but there are other exporters like to csv where this has a big impact (not only in storage space, but also in file space, which incidentally is also where this branch was originally implemented and included into the exporter/importer infrastructure).

Yes the interaction with optional attributes like `p{n}` was not considered at the time, but you should still test out the drawbacks of including everything clearly.
